Src/HALALMock/Services/Logger/Logger.cpp

```diff
@@ -0,0 +1,62 @@
+#include "HALALMock/Services/Logger/Logger.hpp"
+
+#include <ctime>     // To manage time
+#include <format>    // To build formatted strings
+#include <fstream>   // To write in a file
+#include <iostream>  // To write in console
+#include <memory>    // To manage FileManager ptr
+
+// === Class Logger ===
+std::unique_ptr<FileManager> Logger::file_manager =
+    std::make_unique<FileManager>(std::ios_base::trunc);
+int Logger::line = 0;
+const char *Logger::function = nullptr;
+const char *Logger::file = nullptr;
+
+void Logger::log(const std::string &msg, const std::string &level,
+                 const char *colour) {
+    // Format message to include timestamp and Level
+    // Reach timestamp
```

Comment:
comments should be used to explain why something is done, the what is already written there